src/dss/start.py

```diff
@@ -0,0 +1,44 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.autoscaling_v1 import ScaleSpec
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.resources.apps_v1 import Deployment
+
+from dss.config import DSS_NAMESPACE
+from dss.logger import setup_logger
+from dss.utils import does_notebook_exist
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def start_notebook(name: str, lightkube_client: Client) -> None:
+    """
+    Start a Notebook server on the Kubernetes cluster by scaling up the Notebook's Deployment to 1.
+
+    Args:
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    if not does_notebook_exist(
+        name=name, namespace=DSS_NAMESPACE, lightkube_client=lightkube_client
+    ):
+        logger.error(f"Failed to start Notebook. Notebook {name} does not exist.")
+        logger.info("Run 'dss list' to check all notebooks.")
+        raise RuntimeError(f"Failed to start Notebook. Notebook {name} does not exist.")
+
+    obj = Deployment.Scale(
```

Comment:
This, alongside with `dss stop` can affect any `Deployment` in the `DSS_NAMESPACE`. This behaviour is not desired, we would like to only start/stops Notebook resources. The way you can test this is to try to stop `mlflow` after initialising dss `dss stop mlflow  --kubeconfig $KUBECONFIG` and checking the `Deployment`:

```
$ kubectl get deployments -A | grep mlflow
dss               mlflow                    1/1     1            1           10m

```

Then you can start `mlflow` w/o issues:

```
$ dss start mlflow  --kubeconfig $KUBECONFIG
2024-04-19 15:09:26 [INFO] [main] [start_notebook_command]: Executing start command
2024-04-19 15:09:26 [INFO] [start] [start_notebook]: Starting the notebook mlflow. Check `dss list` for the status of the notebook.
```

Reply:
this is a very good point, and I think we will have to change it in this command and all the other commands affected (`remove`, `stop` and possibly `logs`).
I'm filing an issue to track this and tackle it separately.

Reply:
#84

Reply:
Okay, as long as we come back to this.